op-chain-ops: check for protocol version config

tynes opened the pull request:
**Description**

Ensures that the protocol version configs are in the deploy config
by adding them to the `Check()` function to the `DeployConfig` struct.
CI calls this function to ensure that all of the deploy configs have
set values.

<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

Fixes https://github.com/ethereum-optimism/client-pod/issues/76

tynes committed:
op-chain-ops: check for protocol version config

Ensures that the protocol version configs are in the deploy config
by adding them to the `Check()` function to the `DeployConfig` struct.
CI calls this function to ensure that all of the deploy configs have
set values.

protolambda reviewed (approved):
LGTM, although I'd consider the protocol version vars in the deploy config a superchain detail, not a per-chain deployment thing. So maybe we should expand the deploy-config with a variable for a pre-existing superchain-target, and avoid deploying any new superchain things if so.

trianglesphere reviewed (commented):
Need to add the protocol version to at least the devnet deploy config and maybe more deploy configs

mslipper commented:
This is breaking the build - I don't think we're setting the protocol version in the devnet deploy config.

tynes committed:
op-chain-ops: make protocol versions warn instead of err

Instead of erroring it should warn. Otherwise we need to
use a pointer and use `nil` to differentiate from the 0 value.
